docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,552 @@
+      "type": "int"
+    }
+  ]
+}
+----
+
+If you build the project with `mvn compile` , the `movies.avsc` will get compiled to a `Movie.java` file
+placed in the `target/generated-sources/avsc` directory.
+
+If you run the project with `mvn compile quarkus:dev`, the changes you do to the schema file will be
+automatically applied to the generated Java files.
+
+== The `Movie` producer
+Having defined the record, we can now jump to implementing the `MovieResource`.
+
+Let's open the `MovieResource`, inject an `Emitter` of `Movie` DTO and implement a `@POST` method
+that consumes `Movie` and sends it through the `Emitter`:
+
+[source,java]
+----
+package org.acme.kafka;
+
+import org.acme.kafka.quarkus.Movie;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.jboss.logging.Logger;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/movies")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class MovieResource {
+
+    private static final Logger LOGGER = Logger.getLogger(MovieResource.class);
+
+    @Channel("movies") Emitter<Movie> emitter;
+
+    @POST
+    public Response enqueueMovie(Movie movie) {
+        LOGGER.infof("Sending movie %s to Kafka",
+                movie.getTitle()
+        );
+        emitter.send(movie);
+        return Response.accepted().build();
+    }
+
+}
+----
+
+Now, configure the channel for the `Emitter` in `application.properties`:
+[source,properties]
+----
+# set the URL of the Apicurio Schema Registry, a global setting shared between producers and consumers
+mp.messaging.connector.smallrye-kafka.apicurio.registry.url=http://localhost:8081/api
+
+# set the connector to use for the `movies` channel to smallrye-kafka
+mp.messaging.outgoing.movies.connector=smallrye-kafka
+
+# the name of the corresponding Kafka topic to `movies`
+mp.messaging.outgoing.movies.topic=movies
+
+# set the serializer for the `movies` channel to the Avro Serializer for Apicurio
+mp.messaging.outgoing.movies.value.serializer=io.apicurio.registry.utils.serde.AvroKafkaSerializer
+
+# Apicurio schema specific settings:
+mp.messaging.outgoing.movies.apicurio.registry.artifact-id=io.apicurio.registry.utils.serde.strategy.SimpleTopicIdStrategy
+mp.messaging.outgoing.movies.apicurio.registry.global-id=io.apicurio.registry.utils.serde.strategy.GetOrCreateIdStrategy
+mp.messaging.outgoing.movies.apicurio.registry.avro-datum-provider=io.apicurio.registry.utils.serde.avro.ReflectAvroDatumProvider
+----
+
+== The `Movie` consumer
+We already have means to put `Movie` records to a Kafka topic, it's time to implement a consumer for them.
+
+Let's create `ConsumedMovieResource` that will consume `Movie` messages
+from the `movies-from-kafka` channel and will expose it via Server Sent Events:
```

Comment:
```suggestion
from the `movies-from-kafka` channel and will expose it via Server-Sent-Events:
```

Reply:
The canonical name seems to be "Server-sent events" (or "Server-Sent Events" if you want the words capitalized).

I looked at:

- https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events
- https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events
- https://en.wikipedia.org/wiki/Server-sent_events